setup.py

```diff
@@ -48,6 +48,27 @@ def find_files(base_dir, exclude_dirs, include_dirs, file_regex):
     return found
 
 
+def recursive_search(search_list, field):
+    """
+    Takes a list with nested dicts and lists,
+    and searches all dicts for a key of the field
+    provided.
+    """
+    fields_found = []
+
+    for item in search_list:
+        if isinstance(item, dict):
```

Comment:
The docstring says that it takes a list of nested ``dict``s and ``list``s. Wouldn't only checking if the item is a ``dict``cause an early exit?

Reply:
@ashcrow I think what happened here is that I iterated over changes so many times that I ended up having that extra `if` conditional at the beginning.  I'll remove that first statement.  Thanks!

Reply:
And now that I look at it again, it was there because sometimes a `str` was passed into that function.  I'll move the logic for testing for `dict` to the calling statement.

Reply:
@ashcrow Due to the recursive nature of calling `recursive_search`, the list that is passed in could be a list containing `str` as items, such as in the case of `with_items`.  In this case, I don't need to traverse the items and just need to skip over them.  It still makes sense to have this conditional here.  Is there a better way you could suggest?

Reply:
@mtnbikenc I think the conditional is fine, but maybe the docstring needs updating to note if the item being reviewed is not a ``dict`` then the function doesn't check inside the input. Otherwise it sounds like the following would be parsed as well:

```python
[
    {'blah': 'foo', 'bar': 'hi'},
    [{'seemslike': 'idbefound', 're', 'docstring'},
]
```